### PR TITLE
Split some IO precommits out of the Java precommit

### PR DESCRIPTION
**Pull request by kennknowles**
I did this for IOs that already have a precommit job and added a couple others that are notably flaky.

Previously:

 - any change to any IO or Java core would cause unit tests for all IOs to run

Now:

 - any change to Java core causes unit tests for all IOs to run
 - any change to an IO causes only the IOs unit tests to run

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .test-infra/jenkins/PrecommitJobBuilder.groovy [hunk 1]
```diff
@@ -116,7 +119,7 @@ class PrecommitJobBuilder {
       steps {
         gradle {
           rootBuildScriptDir(commonJobProperties.checkoutDir)
-          tasks(gradleTask)
+          tasks(gradleTasks + (gradleTask == null ? []: [gradleTask]))
```
Comment (kennknowles) [inline .test-infra/jenkins/PrecommitJobBuilder.groovy:122]: Note that this change is in a separate commit that should be kept separate for clarity of code history.

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
seed job succeeded this time

**Comment by kennknowles** [comment]
clickhouse and pulsar don't have `integrationTest` targets, so removed those and just kept `build`

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
I actually think that before this change, the cassandra `integrationTest` target was not running in any job. It isn't in the java postcommit.

**Comment by kennknowles** [comment]
Filed #24555. Removed it for now so that this doesn't create a permared job.

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
Run Java_Pulsar_IO_Direct

**Comment by kennknowles** [comment]
Run Java_Clickhouse_IO_Direct

**Comment by kennknowles** [comment]
Run Java_Cassandra_IO_Direct

**Comment by chamikaramj** [comment]
LGTM. Thanks.

> I actually think that before this change, the cassandra integrationTest target was not running in any job
Woops!

**Comment by kennknowles** [comment]
Hmm, the phrases didn't work. But we do have the cron working here: https://ci-beam.apache.org/job/beam_PreCommit_Java_Cassandra_IO_Direct_Cron/3/ and it did trigger on #24410.

**Comment by kennknowles** [comment]
oh it is `Run Java_Cassandra_IO_Direct PreCommit`

**Pull request merged**